src/sql/base/browser/ui/table/plugins/headerFilter.plugin.ts

```diff
@@ -442,10 +441,9 @@ export class HeaderFilter<T extends Slick.SlickData> {
 
 	private async handleMenuItemClick(command: HeaderFilterCommands, columnDef: Slick.Column<T>) {
 		this.hideMenu();
-		const provider = this.grid.getData() as IDisposableDataProvider<T>;
-
-		if (provider.sort && (command === 'sort-asc' || command === 'sort-desc')) {
-			await provider.sort({
+		const dataView = this.grid.getData();
+		if (instanceOfIDisposableDataProvider<T>(dataView) && (command === 'sort-asc' || command === 'sort-desc')) {
```

Comment:
So in some instances we fall back to other behavior if it isn't a IDisposableDataProvider and others (like this one) we don't. Any particular reason for that?

Reply:
agent is using the Slick.Data.DataView as the provider and has a custom sorting logic

https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/jobManagement/browser/jobsView.component.ts#L340

Reply:
Hmm, so if another component was to use this header filter and not use an IDisposbaleDataprovider how are they supposed to know to do this? It just seems a little odd that some stuff is handled by the filter plugin and other stuff isn't. 

anyways not a huge deal - just thought it was a little odd. This stuff is all pretty messy anyways.